playlet-lib/src/components/VideoPlayer/VideoPlayer.bs

```diff
@@ -76,6 +85,9 @@ function PlayWithContent(contentNode as object)
     if not StringUtils.IsNullOrEmpty(contentNode._author)
         videoContentNode.secondaryTitle = contentNode._author
     end if
+    if IsBool(contentNode.liveNow)
+        videoContentNode.live = contentNode.liveNow
+    end if
```

Comment:
A lot of the time this field is missing. For example a content node can come in with just the video Id.
It is best to do this check in OnVideoContentTaskResults where the video details are fetched

Reply:
Also I believe `liveNow` will always be a valid bool on the node, even if it was never set

Reply:
I don't think `live` or `liveNow` is ever propagated to the final content node. Or did you mean something different?

Reply:
Ah, got it, it's in metadata :-)

Reply:
So the contentNode passed to `function PlayWithContent(contentNode as object)` might or might not contain the `liveNow` value. And we're definitely not capturing that here. That's because we will make a request in the `VideoContentTask` Task.

https://github.com/iBicha/playlet/blob/f45ba3097905fa2039b74831ca30f443e3d83611/playlet-lib/src/components/VideoPlayer/VideoContentTask.bs#L43

The contentNode passed to the player will contain a field `metadata` which will be the payload that came from the API. So in the callback `OnVideoContentTaskResults`, `m.top.content.metadata.liveNow` will contain a more accurate value.

To my last point

> Also I believe liveNow will always be a valid bool on the node, even if it was never set

A boolean field on a node will return a default value (false) even if it was never set. So `if IsBool(contentNode.liveNow)` will always return true is what I'm saying

Reply:
> So in the callback OnVideoContentTaskResults, m.top.content.metadata.liveNow will contain a more accurate value.

Yep, figured it right after I sent a previous comment :-D

Reply:
Anyway, I think it works as intended (or at least almost as intended)

With the addition of some data to `playbackActionButtons` (https://developer.roku.com/docs/references/scenegraph/media-playback-nodes/video.md) behavior can be changed to navigate those auxiliary buttons instead and pressing OK would only toggle the HUD visibility.